### PR TITLE
Code changes for serialization & deserialization changes

### PR DESCRIPTION
**Pull request by sarangan12**
Related to https://github.com/Azure/azure-sdk-for-ruby/issues/610

### CODE REVIEW

File: runtime/ms_rest/lib/ms_rest/jsonable.rb [hunk 1]
```diff
@@ -11,17 +11,19 @@ module JSONable
     # Serialize the object to JSON
     # @return [String] JSON serialized version of the object
     #
-    def to_json
-      serialize(self.class.mapper, self)
+    def to_json(mapper = nil)
```
Comment (devigned) [inline runtime/ms_rest/lib/ms_rest/jsonable.rb]: Could we shape this to be named arguments or an options hash similar to Rails?

Comment (devigned) [inline runtime/ms_rest/lib/ms_rest/jsonable.rb]: see http://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html

Comment (sarangan12) [inline runtime/ms_rest/lib/ms_rest/jsonable.rb]: @devigned Done. Can you take a look?

**Comment by sarangan12** [comment]
@veronicagg Please review

**Comment by devigned** [comment]
Awesomesauce!

**Pull request merged**